underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -552,17 +535,63 @@ private boolean isDirectory(String path) throws IOException {
   }
 
   /**
-   * Lists the files or folders which match the given prefix.
+   * Lists the files or folders in the path and remove the path itself.
+   *
+   * @param path the prefix to match
+   * @param recursive whether to do a recursive listing
+   * @return a collection of the files or folders matching the prefix, or null if not found
+   * @throws IOException if path is not accessible, e.g. network issues
+   */
+  private String[] listHelper(String path, boolean recursive) throws IOException {
+    String prefix = addFolderSuffixIfNotPresent(stripContainerPrefixIfPresent(path));
+    prefix = prefix.equals(PATH_SEPARATOR) ? "" : prefix;
+
+    Collection<DirectoryOrObject> objects = listInternal(prefix, !recursive);
+    Set<String> children = new HashSet<>();
+    final String self = stripFolderSuffixIfPresent(prefix);
+    boolean foundSelf = false;
+    for (DirectoryOrObject object : objects) {
+      String child = stripFolderSuffixIfPresent(object.getName());
+      String noPrefix = CommonUtils.stripPrefixIfPresent(child, prefix);
+      if (!noPrefix.equals(self)) {
+        children.add(noPrefix);
+      } else {
+        foundSelf = true;
+      }
+    }
+
+    if (!foundSelf) {
```

Comment:
Could the logic here be simplified by not checking each element and then using `children.contains(self)` as the condition?